### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by nak3**
This patch updates README and Makefile.
It changes:

- to use the `releases/latest/download/kourier.yaml` instead of `releases/download/v0.26.0/kourier.yaml` to keep pointing the latest release.
- to remove Makefile.

/cc @markusthoemmes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,23 +3,11 @@
 SHELL = /bin/bash
 PROJECT_PATH := $(patsubst %/,%,$(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
 
-local-setup: ## Builds and deploys kourier locally in a "kind" cluster with knative, forwards the local 8080 to kourier/envoy
```
Comment (markusthoemmes) [inline Makefile:6]: Is anybody at this point even using `make` here? Should we just get rid of it?

Comment (nak3) [reply Makefile:6]: Yes, let's remove Makefile.

File: README.md [hunk 2]
```diff
@@ -133,29 +133,6 @@ vars in the `net-kourier-controller` deployment:
...
-
-- Set up a local environment with Knative running on top of
-  [kind](https://kind.sigs.k8s.io/) (needs both kind and
-  [ko](https://github.com/google/ko) installed):
-
-```bash
-make local-setup
-```
-
 - Run `make help` for the complete list of make targets available.
```
Comment (markusthoemmes) [inline README.md:136]: Let's drop this as well then :)

Comment (nak3) [reply README.md]: Thank you. Updated.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**